selfdrive/car/mazda/carcontroller.py

```diff
@@ -0,0 +1,91 @@
+from selfdrive.car.mazda import mazdacan
+from selfdrive.car.mazda.values import CAR, DBC
+from selfdrive.can.packer import CANPacker
+from selfdrive.car import apply_std_steer_torque_limits
+
+
+class CarControllerParams():
+  def __init__(self, car_fingerprint):
+    self.STEER_MAX = 600                 # max_steer 2048
```

Comment:
woha, so 2048 must be a lot. How did you pick 600?

Reply:
probably a leftover from subaru

Reply:
Not really, the torque signal is unsigned 12 bit with offset value of 2048 so really the allowed values are [-2048:2047]. I tested values in all of this range and they all worked. The EPS sends out to signals in response to torque signal coming from the camera. Call these ESP_TORQUE which is a copy of the camera torque signal, and the other msg EPS_TORQUE_EFFECTIVE which is the value of the torque actually applied by the EPS. The actual value is capped at 308. i,e range is [-308:308]. The new signal is not a simple trim of the original signal . For example this would be always 0 if torque is not allowed regardless of the value coming from the source signal. It is scaled, smoothed, and trimmed otherwise. I haven't spent enough time to try and decipher the exact translation but I I got the best results  if I limit the max value in the range  of[500 :900] , I think I'm driving with 750 now. higher values seem to decrease steer response due to smoothing I think.  i.e it takes a little longer for the eps to smooth the values when there is huge jumps. I'm assuming Mazda did that to prevent jerky steering. In all cases the value 600 is a very good value. If someone wants to spend time reverse engineer the EPS logic and come up with a better value I'd be happy to use it ;p